src/planscape/forsys/get_forsys_inputs.py

```diff
@@ -143,3 +147,122 @@ def _check_project_area_from_url_params_fields_exist(
             raise Exception('project area field, "polygons" is an empty list')
         if 'id' not in project_area.keys():
             raise Exception('project area missing field, "id"')
+
+
+# Forsys input dataframe headers.
+class ForsysInputHeaders():
+    # Constant headers for project and stand ID's, area, and cost.
+    FORSYS_PROJECT_ID_HEADER = "proj_id"
+    FORSYS_STAND_ID_HEADER = "stand_id"
+    FORSYS_AREA_HEADER = "area"
+    FORSYS_COST_HEADER = "cost"
+
+    # Header prefixes for conditions and priorities.
+    _CONDITION_PREFIX = "c_"
+    _PRIORITY_PREFIX = "p_"
+
+    # List of headers for priorities.
+    # Downstream, this must be in the same order as constructor input
+    # priorities.
+    priority_headers: list[str]
+
+    def __init__(self, priorities: list[str]) -> None:
+        self.priority_headers = []
+
+        for p in priorities:
+            self.priority_headers.append(self.priority_header(p))
+
+    # Returns a priority header givn a priority string.
+    def priority_header(self, priority: str) -> str:
+        return self._PRIORITY_PREFIX + priority
+
+    # Reteurns a condition hader given a condition string.
+    def condition_header(self, condition: str) -> str:
+        return self._CONDITION_PREFIX + condition
+
+
+class ForsysProjectAreaRankingInput():
+    # dictionary representing a forsys input dataframe.
+    forsys_input: dict[str, list]
+
+    def __init__(
+            self, params: ForsysProjectAreaRankingRequestParams,
+            headers: ForsysInputHeaders) -> None:
+        region = params.region
+        priorities = params.priorities
+        project_areas = params.project_areas
+
+        condition_ids_to_names = self._get_condition_ids_to_names(
+            region, priorities)
+        conditions = self._get_conditions(condition_ids_to_names.keys())
+
+        self.forsys_input = self._initialize_forsys_input(headers, priorities)
+
+        for proj_id in project_areas.keys():
+            geo = self._get_raster_geo(project_areas[proj_id])
+
+            self.forsys_input[headers.FORSYS_PROJECT_ID_HEADER].append(proj_id)
+            self.forsys_input[headers.FORSYS_STAND_ID_HEADER].append(proj_id)
```

Comment:
i don't think i have enough context to review this particular change. @JanLauGe is OOO atm, is it possible to separate this change into a diff PR? that way the changes can be more modular.

Reply:
It's hard to make all the other changes and keep this as it was.

Not checking this in will also block follow-up work.

Just for additional context:
Previously, the input included multiple stands within each project area. Now, the input has a single stand within each project area. This does affect the stand output, but doesn't affect the project output. Since we're working on ranking at the project level, we only care about the project output. In sum, this change reduces the number of inputs without affecting the output.

Reply:
thanks for the explanation! can we include that in the top level description of the PR?

Reply:
yes - included.